providers/implementations/macs/kmac_prov.c

```diff
@@ -340,6 +360,14 @@ static int kmac_final(void *vmacctx, unsigned char *out, size_t *outl,
     if (!ossl_prov_is_running())
         return 0;
 
+#ifdef FIPS_MODULE
+    if (FIPS_kmac_length_checks_enabled(PROV_LIBCTX_OF(kctx->provctx))
+            && kctx->out_len < KMAC_MIN_OUTPUT_LEN_FIPS) {
```

Comment:
Can this actually happen if we have the check in `kmac_set_ctx_params`?

Reply:
No.  There is another error check missing however.